### PR TITLE
indexer-cli: indexing rule parser error catch

### PR DESCRIPTION
**Pull request by hopeyen**
When using wrong arguments for `indexer rules set`, the error message 
`TypeError: INDEXING_RULE_PARSERS[key] is not a function` 
is not very clear. 

Improve the error message to specify the invalid key: when provided a non-existing attribute, return messages like
`Indexing rule attribute 'allocatoinAmount' doesn't exist, please check spelling`

for next step: can make suggestions using modules like `spellchecker-cli`

### CODE REVIEW

**Comment by fordN** [comment]
Nice! Can we also output the help text after the message so the user can see what the supported parameters are?

**Comment by hopeyen** [comment]
> Nice! Can we also output the help text after the message so the user can see what the supported parameters are?

I ended up making a simple dumb suggestion function: if there's valid commands that shares 80% of the inputted letters, print those, otherwise print all available commands. Let me know if you think the help text would be better?

File: packages/indexer-common/src/indexer-management/models/indexing-rule.ts [hunk 1]
```diff
@@ -30,6 +30,24 @@ export interface IndexingRuleAttributes {
   requireSupported: boolean
 }
 
+export const IndexingRuleAttributeDictionary = [
```
Comment (fordN) [inline packages/indexer-common/src/indexer-management/models/indexing-rule.ts:33]: I don't expect the indexing rule models to change, but in order to ease management and make the code more "DRY" can we avoid creating a new constant here and instant derive the array of keys from the `IndexingRuleAttributes` interface or the `IndexingRule` class?

File: packages/indexer-cli/src/rules.ts [hunk 2]
```diff
@@ -137,7 +137,14 @@ export const parseIndexingRule = (
   const obj = {} as any
   for (const [key, value] of Object.entries(rule)) {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    obj[key] = (INDEXING_RULE_PARSERS as any)[key](value)
+    const parser = (INDEXING_RULE_PARSERS as any)[key]
+    if (!parser) {
+      throw (
+        `Indexing rule attribute '${key}' doesn't exist, please check spelling, might be one of these:\n` +
```
Comment (fordN) [inline packages/indexer-cli/src/rules.ts:143]: What do you think about changing the text to this? 
`Indexing rule attribute '${key}' not supported, did you mean?`

File: packages/indexer-common/src/indexer-management/models/indexing-rule.ts (hunk 1 above)
Comment (hopeyen) [reply packages/indexer-common/src/indexer-management/models/indexing-rule.ts:33]: I totally agree! 'Instant derive' is what I was missing. Do you think we could use module `ts-transformer-keys`?

Comment (fordN) [reply packages/indexer-common/src/indexer-management/models/indexing-rule.ts:33]: I think this should be doable without any additional libraries.  

Perhaps we can use the rawAttributes property of the Sequelize model?
`Object.keys(IndexingRule.rawAttributes)`?

or use the keys from an instance of the class?
`Object.keys(new IndexingRule())`?

**Comment by fordN** [comment]
@hopeyen what's the status on this PR? Ready for another review?

**Comment by hopeyen** [comment]
> 

yep this one is ready for another review! thanks in advance

File: packages/indexer-cli/src/commands/indexer/rules/set.ts [hunk 3]
```diff
@@ -61,23 +63,31 @@ module.exports = {
...
+            'Parallel allocations are soon to be fully deprecated. Please set parallel allocations to 1 for all your indexing rules',
+          )
+          process.exitCode = 1
+        }
+        const rule = await setIndexingRule(client, inputRule)
+        printIndexingRules(print, outputFormat, identifier, rule, [])
+      } catch (error) {
+        // Failed to parse input, make suggestions
+        const valid_commands = Object.keys(
+          (await indexingRules(client,false))[0],
```
Comment (hopeyen) [inline packages/indexer-cli/src/commands/indexer/rules/set.ts:84]: or I could call `await indexingRule(client,'global', false)`? although that returns type potentially null that needs more conditionals to handle. Getting index 0 here is basically the global rule which is ensured to exist. Not sure if we need to worry about not having an empty list

**Review by fordN — APPROVED** [review]
LGTM 👍

**Pull request merged**